lib/core/mqtt.hpp

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "logger.hpp"
+#include "mqtt_topics.hpp"
+#include "types.hpp"
+
+#include <optional>
+#include <queue>
+#include <vector>
+
+#include <mqtt/client.h>
+#include <rapidjson/document.h>
+
+namespace hyped::core {
+
+enum MqttMessagePriority {
+  kCritical = 0,
+  kNormal   = 1,
+  kLow      = 2,
+};
+
+enum MqttMessageQos {
+  kAtMostOnce  = 0,
+  kAtLeastOnce = 1,
+  kExactlyOnce = 2,
+};
+
+struct MqttMessage {
+  MqttTopic topic;
+  struct Header {
+    std::uint64_t timestamp;
+    MqttMessagePriority priority;
+  };
+  Header header;
+  rapidjson::Document &payload;
```

Comment:
Problem with the assignment of the payload member of the struct due it being a reference. I just added a constructor of the form
```
MqttMessage(MqttTopic topic, Header& header, rapidjson::Document &payload) : topic(topic), 
    header(header), 
    payload(payload) {};
```
to fix the assignment issue.